objectivec/GPBWellKnownTypes.h

```diff
@@ -96,4 +115,82 @@ NS_ASSUME_NONNULL_BEGIN
+ */
++ (nullable instancetype)anyWithMessage:(nonnull GPBMessage *)message
+                                  error:(NSError **)errorPtr;
+
+/**
+ * Returns a new instance of GPBAny that includes the serialized message. On
+ * failure returns nil and fills in the optional errorPtr with the details. The
+ * error can be a GPBWellKnownTypesErrorDomain error or a general message
+ * serialization error.
+ */
++ (nullable instancetype)anyWithMessage:(nonnull GPBMessage *)message
+                          typeURLPrefix:(nonnull NSString *)typeURLPrefix
+                                  error:(NSError **)errorPtr;
+
+/**
+ * Serializes message and initializes the instance to contain the info. Uses
+ * the type.googleapis.com/ type_url prefix.  On failure returns nil and fills
+ * in the optional errorPtr with the details. The error can be a
+ * GPBWellKnownTypesErrorDomain error or a general message serialization error.
+ */
+- (nullable instancetype)initWithMessage:(nonnull GPBMessage *)message
+                                   error:(NSError **)errorPtr;
+
+/**
+ * Serializes message and initializes the instance to contain the info. On
+ * failure returns nil and fills in the optional errorPtr with the details. The
+ * error can be a GPBWellKnownTypesErrorDomain error or a general message
+ * serialization error.
+ */
+- (nullable instancetype)initWithMessage:(nonnull GPBMessage *)message
+                           typeURLPrefix:(nonnull NSString *)typeURLPrefix
+                                   error:(NSError **)errorPtr;
+
+/**
+ * Serializes message and sets the current instance to contain the info. Uses
+ * the type.googleapis.com/ type_url prefix. Returns success/failure, and on
+ * failure fills in the optional errorPtr with the details. The error can be a
+ * GPBWellKnownTypesErrorDomain error or a general message serialization error.
+ */
+- (BOOL)packWithMessage:(nonnull GPBMessage *)message
+                  error:(NSError **)errorPtr;
+
+/**
+ * Serializes message and sets the current instance to contain the info.
+ * Returns success/failure, and on failure fills in the optional errorPtr the
+ * details. The error can be a  GPBWellKnownTypesErrorDomain error or a general
+ * message serialization error (see GPBMessage).
+ */
+- (BOOL)packWithMessage:(nonnull GPBMessage *)message
+          typeURLPrefix:(nonnull NSString *)typeURLPrefix
+                  error:(NSError **)errorPtr;
+
+/**
+ * Checks if the type_url matches the requested class and then parses the data
+ * returning the message on success. On failure nil is returned and the
+ * optional errorPtr will be filled in; the error could be a
+ * GPBWellKnownTypesErrorDomain error or a general message parsing error
+ * (see GPBMessage and GPBCodedInputStream).
+ * @note When checking type_url, the base url isn't checked, only the
+ * package.message part.
+ */
+- (nullable GPBMessage *)unpackMessageClass:(Class)messageClass
```

Comment:
Unpacks the serialized message as if it was an instance of the given class.

@note When checking type_url, the base URL is not checked, only the fully qualified name.

@ param messageClass The class to use to deserialize the contained message.
@ param errorPtr Pointer to an error that will be populated if something goes wrong.

@ return An instance of the given class populated with the contained data, or nil on failure.